base/js/admin.js

```diff
@@ -615,19 +616,22 @@
                     var $in = $(input);
 
                     if(typeof pos !== 'undefined') {
-                        var newName = $in.data('original-name');
+                        var newName = $in.attr('data-original-name');
```

Comment:
The difference is that the attribute survives across duplication. There was a problem when duplicating a repeater item, for the widget settings form. Oddly, I wasn't getting the same issue when using the form in a standard widget form though.